File: go/muse/util/RandomFactory/RandomFactory.go
package RandomFactory

import (
	"crypto/rand"
	"math/big"
)

func GenIntN(min int, max int) int {
	n, _ := rand.Int(rand.Reader, big.NewInt(int64(max-min+1)))
	return min + int(n.Int64())
}

func GenInt() int {
	return GenIntN(1, 2147483647)
}

func GenEven() int {
	return GenInt() >> 1 << 1
}

func GenOdd() int {
	return GenInt() | 1
}

File: archive/nodejs/muse/util/RandomFactory.js
import * as crypto from 'node:crypto';

export const genIntN = (min, max) => {
  return crypto.randomInt(min, max + 1);
};

export const genInt = () => {
  return genIntN(1, 2_147_483_647);
};

export const genEven = () => {
  return (genInt() >>> 1) << 1;
};

export const genOdd = () => {
  return genInt() | 1;
};
